### PR TITLE
Expose run information as environmental variables

### LINKED ISSUE #4904: Expose run information as environmental variables
We have users who are parsing the runpath to get metadata about the ert run. This is data that we have available in ert, and we should provide it in a more robust fashion,

Environment variables on client:
1) Experiment ID
1) Realization ID
1) Simulation mode

On compute:
1) Iteration number
1) Realization number
1) Runpath format string

### PR DESCRIPTION
**Pull request by DanSava**
**Issue**
Resolves #4904 

**Approach**
_Short description of the approach_

## Pre review checklist

- [ ] Added appropriate release note label
- [ ] PR title captures the intent of the changes, and is fitting for release notes.
- [ ] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).
- [ ] Updated documentation
- [ ] Ensured new behaviour is tested

Adding labels helps the maintainers when writing release notes. This is the [list of release note labels](https://github.com/equinor/ert/labels?q=release-notes).

### CODE REVIEW

**Comment by DanSava** [comment]
I am a bit worried because this PR essentially makes the keys `ENV` and `EXEC_ENV` the same once they are defined in a job config file. 

I could not find any substantial documentation that would explain how these keys were designed to work but I would assume that was supposed to be a difference between them otherwise there would not be two keys doing the same thing.

**Comment by DanSava** [comment]
Looked through the logs and the only config file which uses `EXEC_ENV` is for the old_stlye RMS job which looks like the following:
```
EXECUTABLE ../res/script/rms.py

DEFAULT <RMS_IMPORT_PATH> ./
DEFAULT <RMS_EXPORT_PATH> ./
DEFAULT <RMS_RUNPATH> rms/model
DEFAULT <RMS_OPTS> ""
TARGET_FILE <RMS_TARGET_FILE>
ARGLIST <IENS> <RMS_PROJECT> <RMS_WORKFLOW> -r <RMS_RUNPATH> -t <RMS_TARGET_FILE> -i <RMS_IMPORT_PATH> -v <RMS_VERSION> -e <RMS_EXPORT_PATH> <RMS_OPTS>

EXEC_ENV PYTHONPATH <RMS_PYTHONPATH>
EXEC_ENV PATH_PREFIX <RMS_PATH_PREFIX>
```
Checking for `RMS_PATH_PREFIX` in the logs it seems it is never defined in any ert config file and this leads to the warning 
`Environment variable PATH_PREFIX skipped due to unmatched define <RMS_PATH_PREFIX>` logged `384784` times So I guess this can be removed from the RMS config file

but the `RMS_PYTHONPATH` is used in some config files and the rms.py job expects this variable to be written to `rms_exec_env.json` and does 'something' with it not yet clear. 

Could not find any instance of usage for the `ENV` keyword

File: src/_ert_job_runner/job.py [hunk 1]
```diff
@@ -76,23 +76,17 @@ def run(self):
...
-            )
-            with open(f"{exec_name}_exec_env.json", "w", encoding="utf-8") as f_handle:
-                f_handle.write(json.dumps(exec_env, indent=4))
-
         max_running_minutes = self.job_data.get("max_running_minutes")
         run_start_time = dt.now()
-
+        environment = self.job_data.get("environment")
+        if environment is not None:
+            environment = {**os.environ, **environment}
```
Comment (andreas-el) [inline src/_ert_job_runner/job.py:83]: So if the environment is not set, we won't pass os.environ?

File: tests/unit_tests/c_wrappers/job_runner/test_jobmanager.py [hunk 2]
```diff
@@ -187,4 +183,85 @@ def test_exec_env():
             with open("exec_env_exec_env.json", encoding="utf-8") as f:
                 exec_env = json.load(f)
                 assert exec_env["TEST_ENV"] == "123"
-                assert exec_env["NOT_SET"] is None
+        if isinstance(msg, Exited):
+            assert msg.exit_code == 0
+
+
+@pytest.mark.usefixtures("use_tmpdir")
+def test_job_env():
```
Comment (andreas-el) [inline tests/unit_tests/c_wrappers/job_runner/test_jobmanager.py:191]: Should expand the test name to convey full intent of the test.

File: tests/unit_tests/c_wrappers/job_runner/test_jobmanager.py [hunk 3]
```diff
@@ -187,4 +183,85 @@ def test_exec_env():
...
+    for msg in list(JobRunner(jobs_json).run([])):
+        if isinstance(msg, Exited):
+            assert msg.exit_code == 0
+
+    # Check ENV variable not available outside of job context
+    assert "TEST_ENV" not in os.environ
+
+
+@pytest.mark.usefixtures("use_tmpdir")
+def test_job_default_env():
```
Comment (andreas-el) [inline tests/unit_tests/c_wrappers/job_runner/test_jobmanager.py:226]: Same as above

File: tests/unit_tests/c_wrappers/job_runner/test_jobmanager.py [hunk 4]
```diff
@@ -187,4 +183,85 @@ def test_exec_env():
...
+    for k in ExtJob.default_env:
+        assert k not in os.environ
+
+    for msg in list(JobRunner(jobs_json).run([])):
+        if isinstance(msg, Exited):
+            assert msg.exit_code == 0
+
+    # Check default ENV variable not available outside of job context
+    for k in ExtJob.default_env:
+        assert k not in os.environ
```
Comment (andreas-el) [inline tests/unit_tests/c_wrappers/job_runner/test_jobmanager.py:267]: Am i correct in  that there is a hidden verification in here, -- you indirectly check that the default environment set in the python script is present, and after jobs have finished, you verify that the os.environment is not polluted?

**Review by andreas-el — CHANGES_REQUESTED** [review]
Some minor requests to change naming of tests. It is not clear what's actually verified until you inspect the test-code.
Otherwise well done. :+1:

File: src/_ert_job_runner/job.py (hunk 1 above)
Comment (DanSava) [reply src/_ert_job_runner/job.py:83]: yes

File: tests/unit_tests/c_wrappers/job_runner/test_jobmanager.py (hunk 4 above)
Comment (DanSava) [reply tests/unit_tests/c_wrappers/job_runner/test_jobmanager.py:267]: Yes, that is correct.

**Review by andreas-el — APPROVED** [review]
Good work sir! :rocket:

**Pull request merged**